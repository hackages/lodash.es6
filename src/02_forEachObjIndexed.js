/**
 * Iterate over an input object, calling a provided function callback for each key and value in the object.
 **/
export function forEachObjIndexed(object, callback) {}
