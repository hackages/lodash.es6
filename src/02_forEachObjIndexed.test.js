import {forEachObjIndexed} from './02_forEachObjIndexed'

describe('forEachObjIndexed', () => {
  it('should map values of an object', () => {
    const result = []
    const results2 = []
    let name = ''
    let count = 0

    forEachObjIndexed({a: 33}, (a, b) => result.push([a, b]))

    forEachObjIndexed({a: undefined, b: null, c: true}, (a, b) =>
      results2.push([a, b])
    )

    forEachObjIndexed(
      {4: 42, 35: 35},
      (a, b) => (count += parseInt(a) + parseInt(b))
    )

    forEachObjIndexed({name: 'Antonio'}, (a, b) => (name = `My ${b} is ${a}`))

    expect(result).toMatchSnapshot()
    expect(results2).toMatchSnapshot()
    expect(count).toMatchSnapshot()
    expect(name).toMatchSnapshot()
  })
})
