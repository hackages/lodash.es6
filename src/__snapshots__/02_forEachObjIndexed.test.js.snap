// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`forEachObjIndexed should map values of an object 1`] = `
Array [
  Array [
    33,
    "a",
  ],
]
`;

exports[`forEachObjIndexed should map values of an object 2`] = `
Array [
  Array [
    undefined,
    "a",
  ],
  Array [
    null,
    "b",
  ],
  Array [
    true,
    "c",
  ],
]
`;

exports[`forEachObjIndexed should map values of an object 3`] = `116`;

exports[`forEachObjIndexed should map values of an object 4`] = `"My name is Antonio"`;
